resourseleak/TestConfidenceScores.java

```diff
@@ -0,0 +1,114 @@
+import com.amazonaws.a4b.endpointoperationservice.util.SerializationUtils;
+import com.amazonaws.a4b.platform.common.exception.HttpClientException;
+import com.amazonaws.a4b.platform.exception.InvalidDeviceStateException;
+import com.amazonaws.a4b.platform.utils.HttpClientHelper;
+import com.amazonaws.services.s3.model.S3Object;
+import org.apache.commons.io.IOUtils;
+
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.time.Clock;
+
+public class TestConfidenceScores {
+
+    private static final int CONNECTION_TIME_OUT = 3000;
+    private static final String APP_NAME = "A4BAnalyticsLambda";
+    private final MediaStorageServiceAccessor mssAccessor;
+    private final S3Logic s3Logic;
+    private final OperationLogic operationLogic;
+    private final Clock clock;
+
+
+    public String testConfidenceScores1() throws IOException {
+        final StringWriter writer = new StringWriter();
+        IOUtils.copy(this.getClass().getResourceAsStream("/report_template.html"), writer);
+        return writer.toString();
+    }
+
+    public RegisterAVSDeviceResponse testConfidenceScores2(final String clientId, final String customerId, final String productId,
+                                                       final String productSerialNumber) {
+
+        try {
+            final String adrsRegisterAVSDeviceEndpoint = adrsEndpoint + ADRS_REGISTER_AVS_DEVICE_PATH;
+
+            final WebTarget target = client.target(adrsRegisterAVSDeviceEndpoint).register(AaaFeature.class).queryParam("clientId", clientId)
+                    .queryParam("accountPool", accountPool).queryParam("productSerialNumber", productSerialNumber).queryParam("customerId", customerId)
+                    .queryParam("productId", productId);
+            final Response response = target.request().property(SERVICE_KEY, SERVICE_NAME).property(OPERATION_KEY, OPERATION_NAME).get();
+
+            HttpClientHelper.inspectStatusCode(response, SERVICE_NAME);
+            return response.readEntity(RegisterAVSDeviceResponse.class);
+        } catch (HttpClientException e) {
+            throw new InvalidDeviceStateException(ADRS_ERROR_MESSAGE, e);
+        } catch (DependencyServiceException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new DependencyServiceException(ADRS_ERROR_MESSAGE, e);
+        }
+    }
+
+    private static final File testConfidenceScores3(final String str, final String prefix, final String suffix) {
+        final File tempFile;
+
+        try {
+            tempFile = File.createTempFile(prefix, suffix);
+        } catch (IOException e1) {
+            throw new AmazonServiceException("Could not create temp file, temp dir: " +
+                    System.getProperty("java.io.tmpdir"));
+        }
+
+        try {
+            final FileWriter fw = new FileWriter(tempFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced in statements at the following line: 78. The resource closure statement is at line: 79. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Either a) close *fw* in a try-finally block or b) close the resource by declaring *fw* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).